# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1425626761 -3600
#      Fri Mar 06 08:26:01 2015 +0100
# Node ID bb1629e15afeb12d67966985206c3171da73703b
# Parent  06979876af7daf67befb63785c3518b134a9014e
cancel run action when saving unnamed file was canceled (bug #44434)

* file-editor-tab.cc (run_file): check for success of saving an unamed file

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -703,17 +703,21 @@ file_editor_tab::print_file (const QWidg
 
 void
 file_editor_tab::run_file (const QWidget *ID)
 {
   if (ID != this)
     return;
 
   if (_edit_area->isModified () | ! valid_file_name ())
-    save_file (_file_name);  // save file dialog
+    {
+      save_file (_file_name);  // save file dialog
+      if (! valid_file_name ())
+        return;   // still invalid file name: "save as" was cancelled
+    }
 
   QFileInfo info (_file_name);
   emit run_file_signal (info);
 }
 
 void
 file_editor_tab::context_run (const QWidget *ID)
 {
